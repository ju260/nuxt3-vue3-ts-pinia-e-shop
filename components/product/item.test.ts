import Component from './item.vue';
import itemData from '@/components/Product/__fixtures__/item.js';
import ProductItemImage from '@/components/Product/item-image.vue';
import ProductItemPrice from '@/components/Product/itemPrice.vue';
import { mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { describe, it, expect, beforeEach } from 'vitest';

describe('ProductItem', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Component, {
      shallow: true,
      global: {
        plugins: [createTestingPinia()],
      },
      propsData: {
        item: itemData,
      },
      stubs: {
        'Product-item-image': ProductItemImage,
        'Product-item-price': ProductItemPrice,
      },
    });
  });

  describe('render', () => {
    it('should render component ', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('adds quantity', async () => {
    await wrapper.find('[data-test="addQty"]').trigger('click');
    expect(wrapper.vm.itemQty).toBe(3);
  });

  it('removes quantity', async () => {
    await wrapper.find('[data-test="removeQty"]').trigger('click');
    expect(wrapper.vm.itemQty).toBe(1);
  });

  it('deletes an item', async () => {
    const spyRemoveItem = vi.spyOn(wrapper.vm, 'removeItem');
    await wrapper.find('[data-test="delete"]').trigger('click');
    expect(spyRemoveItem).toHaveBeenCalledWith('166174');
  });
});
