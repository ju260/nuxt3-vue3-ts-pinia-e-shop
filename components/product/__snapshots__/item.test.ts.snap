// Vitest Snapshot v1

exports[`ProductItem > render > should render component  1`] = `
<div
  class="border-b-1 text-left"
>
  <article
    class="flex"
  >
    <div
      class="relative mr-2 flex w-7/12 justify-center"
    >
      <product-item-image
        class="absolute max-h-full object-contain"
        srcset-obj="[object Object]"
      />
    </div>
    <div>
      <p>
        <span
          class="text-base underline"
        >
          ref: 166174
        </span>
      </p>
      <h3>
        Canap√©-lit style scandinave 3 places gris clair
      </h3>
      <p>
        <product-item-price
          item="[object Object]"
        />
      </p>
      <div
        class="flex"
      >
        <button
          class="rounded border border-blue-500 bg-transparent py-1 px-1 font-semibold text-blue-700 hover:border-transparent hover:bg-blue-500 hover:text-white"
          data-test="removeQty"
        >
           - 
        </button>
        <input
          class="w-10 text-center"
          disabled=""
          min="1"
          type="number"
        />
        <button
          class="rounded border border-blue-500 bg-transparent py-1 px-1 font-semibold text-blue-700 hover:border-transparent hover:bg-blue-500 hover:text-white"
          data-test="addQty"
        >
           + 
        </button>
      </div>
      <button
        class="mt-2 rounded bg-blue-500 py-2 px-4 font-bold text-white hover:bg-blue-700"
        data-test="delete"
      >
         Supprimer 
      </button>
    </div>
  </article>
</div>
`;
